app/front/scripts/components/babbage-geoview/utils.js

```diff
@@ -76,35 +80,84 @@ function updateDimensions(geoJson, options) {
   geoJson.unscaledBounds = path.bounds(geoJson);
 }
 
-function updateValues(geoJson, options) {
-  var sum = 0;
+// This algorithm is used to find min and max value for coloring range.
+// Values array may contain several values that are 100 or more times larger
+// then other values. In this case, we'll have few fully colored regions
+// on map, and other ones will be desaturated (because all of them has
+// 100 times less values, comparing to max). So this function applies
+// median filter with adaptive window to array of values, and then takes min
+// and max values. This makes min/max range much closer to most values in array.
+function findMedianRange(values) {
   var min = null;
   var max = null;
+  var radius = Math.ceil(values.length / 5);
+  if (radius > 10) {
+    radius = 10;
+  }
+
+  if (values.length > 0) {
+    _.each(values, function(value, index) {
+      var sum = 0;
+      var j = 0;
+      for (var i = index - radius; i <= index + radius; i++) {
+        j = i;
+        // Normalize index to array bounds
+        if (j < 0) {
+          j = values.length + j;
+        }
+        if (j >= values.length) {
+          j = j - values.length;
+        }
+        // Use value
+        sum += values[j];
+      }
+      sum = sum / (2 * radius + 1);
+      if ((min === null) || (sum < min)) {
+        min = sum;
+      }
+      if ((max === null) || (sum > max)) {
+        max = sum;
+      }
+    });
+  }
+
+  var result = [min, max];
+  result.min = min;
+  result.max = max;
+  return result;
+}
+
+function updateValues(geoJson, options) {
+  var values = [];
   _.each(geoJson.features, function(item) {
     item.value = options.data[item.properties.name];
     if (!_.isUndefined(item.value)) {
-      sum += item.value;
-      if ((min === null) || (item.value < min)) {
-        min = item.value;
-      }
-      if ((max === null) || (item.value > max)) {
-        max = item.value;
-      }
+      values.push(item.value);
     }
   });
+  var range = findMedianRange(values);
 
   var scale = d3.scale.linear()
-    .domain([min, max])
+    .domain(range)
     .range([0, 1]);
```

Comment:
question: 
what if a single value is out of the domain? 
should you be using `d3.clamp` or is it ok for the resulting value to be outside of the 0..1 range?

(https://github.com/mbostock/d3/wiki/Quantitative-Scales#linear_clamp)